src/render.js

```diff
@@ -71,7 +70,9 @@ const normalizeMapboxSourceURL = (url, token) => {
         urlObject.query.access_token = token
         return urlLib.format(urlObject)
     } catch (e) {
-        throw new Error(`Could not normalize Mapbox source URL: ${url}\n${e}`)
+        const msg = `Could not normalize Mapbox source URL: ${url}\n${e}`
+        logger.error(msg)
+        throw new Error(msg)
```

Comment:
I assume there's no simple way to catch and log errors at a higher level?

Reply:
Yes and no.  Commit 42184b3ecc7a3f0d5d961c4c6a9c10f91531dab3 fixed the error logging at the server level (server must be started in verbose mode via `-v`); previously it was it was not passing back the string error message correctly.  Fixing this allowed us to reduce the double-logging in the server tier.

Within the renderer it is a bit of a different story: there are some things that we can propagate back up via exception handling (by throwing errors) whereas in other places these are outside the main control flow.  For example, we cannot raise exceptions based on style parsing error messages / warnings passed back by maplibre, since these happen outside the normal control flow (this is a [known issue](https://github.com/maplibre/maplibre-native/issues/929)).

I did remove some of the unnecessary log statements from `renderer` related to validating inputs; those are already validated in the server tier, and when using the NodeJS API directly it is OK for those to simply raise errors.